compiler-core/src/parse.rs

```diff
@@ -2176,6 +2195,22 @@ where
         } else {
             (vec![], end)
         };
+
+        // check if all constructors are deprecated if so error
+        if constructors.len() > 0 // prevent checking an empty type `type wobble` which will always be true
+            && constructors
+                .iter()
+                .all(|record| record.deprecation.is_deprecated())
+        {
+            return parse_error(
+                ParseErrorType::AllVariantRecordConstructorDeprecated,
+                SrcSpan {
+                    start,
+                    end: end_position,
```

Comment:
Wondering here whether we include the whole type or just up to the end of the type name?

```
  ┌─ /src/parse/error.gleam:2:1
  │  
2 │ ╭ type Wibble {
3 │ │     @deprecated("1")
4 │ │     Wibble1
5 │ │     @deprecated("2")
6 │ │     Wibble2
7 │ │ }
  │ ╰─^ Can't deprecate all variants of a type.
```


```
  ┌─ /src/parse/error.gleam:2:1
  │  
2 │ type Wibble {
  │ ^^^^^^^^^^^ Can't deprecate all variants of a type.
```

Reply:
decided to emit the latter--single-line error. Considering if a custom type might be too and redundant to print.